File: Python/Projects/Cells/cells.py
import math
import sys
import random


def solve():
    f = sys.stdin
    while True:
        n = int(f.readline())
        if (n == 0):
            return
        center = tuple(map(float, f.readline().split()))
        towers = [tuple(map(float, f.readline().split())) for i in range(n-1)]
        getCoverage(center, towers)


def getCoverage(center, towers):
    count = total = 0
    
    for i in range(100000):
        x,y,r = center
        rx = (random.random() * r * 2) - r + x
        ry = (random.random() * r * 2) - r + y
        
        # Check that point is actually inside the town circle
        if not isInCircle(x, y, r, rx, ry):
            continue
        
        for tower in towers:
            x,y,r = tower
            if isInCircle(x, y, r, rx, ry):
                count += 1
                break
        total += 1

    print "{0:.2f}".format(float(count) / total)

        
def isInCircle(x1, y1, r, x2, y2):
    return math.sqrt(((x1 - x2) ** 2) + ((y1 - y2) ** 2)) <= r


if __name__ == '__main__':
    solve()

File: Java/ProgrammingTeam/CellsJava/Cells.java
import java.util.Random;
import java.util.Scanner;

public class Cells 
{
	public static void main(String[] args) 
	{
		Scanner in = new Scanner(System.in);
		while (in.hasNextDouble()) {
			int n = in.nextInt();
			if (n == 0)
				break;
			
			Circle center = new Circle(in.nextDouble(), in.nextDouble(), in.nextDouble());
			Circle[] towers = new Circle[n - 1];
			for (int i = 0; i < n -1; i++) {
				towers[i] = new Circle(in.nextDouble(), in.nextDouble(), in.nextDouble());
			}
			
			System.out.printf("%.2f\n", getCoverage(center, towers));
		}
		in.close();
	}
	
	
	public static double getCoverage(Circle c, Circle[] towers)
	{
		int count = 0, total = 0;
		Random r = new Random();
		for (int i = 0;i < 100000; i++) {
			// Generate random point
			double x = (r.nextDouble() * c.radius * 2) - c.radius + c.x;
			double y = (r.nextDouble() * c.radius * 2) - c.radius + c.y;
			
			// Check if the point's coordinates are in the town circle
			if (!isInCircle(c, x, y))
				continue;
			
			// Find a tower whose coverage includes the point
			for (Circle tower : towers) {
				if (isInCircle(tower, x, y)) {
					count++;
					break;
				}
			}
			total++;
		}
		
		return (count * 1.0) / total;
	}
	
	
	public static boolean isInCircle(Circle c, double x, double y)
	{
		return Math.sqrt(Math.pow(c.x - x, 2) + Math.pow(c.y - y, 2)) <= c.radius;
	}
	
	
	public static class Circle
	{
		public double x;
		public double y;
		public double radius;
		
		public Circle(double x, double y, double radius) {
			this.x = x; this.y = y; this.radius = radius;
		}
	}
}
